core/include/macros/math.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2022 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     core_macros
+ * @{
+ *
+ * @file
+ * @brief       Math helper macros
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ */
+
+#ifndef MACROS_MATH_H
+#define MACROS_MATH_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief internal helper function
+ * @param[in] a integer
+ * @return -1 if @p is negative
+ *          1 otherwise
+ */
+static inline int __signof(long long a)
+{
+    return a < 0 ? -1 : 1;
+}
+
+/**
+ * @brief Returns the sign of @p a, either -1 or 1
+ */
+#define SIGNOF(a) _Generic(a, unsigned char: 1,          \
+                              unsigned short: 1,         \
+                              unsigned int: 1,           \
+                              unsigned long: 1,          \
+                              unsigned long long: 1,     \
+                              default: __signof(a))
```

Comment:
```suggestion
                              default: (a) < 0 ? -1 : 1)
```

? `__singof()` is only used here anyway and this way `clangd` won't nag me about reserved identifiers being used.